src/test/java/core/basesyntax/service/ReaderServiceImplTest.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.service.impl.ReaderServiceImpl;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public class ReaderServiceImplTest {
+    private final ReaderService readerService = new ReaderServiceImpl();
+
+    @Test
+    void read_validFile_notOk() throws IOException {
+        Path tempFile = Files.createTempFile("test", ".txt");
```

Comment:
it seems to me that the file should have a csv extension
the same in other cases